pkg/controller/vitessshard/reconcile_tablets.go

```diff
@@ -164,6 +168,7 @@ func (r *ReconcileVitessShard) reconcileTablets(ctx context.Context, vts *planet
 			newObj := obj.(*corev1.Pod)
 			tablet := tabletMap[key]
 			vttablet.UpdatePodInPlace(newObj, tablet)
+			newObj.Annotations[observedShardGenerationAnnotationKey] = strconv.FormatInt(vts.Generation, 10)
```

Comment:
Any time we directly set an item in this map, we need to check if we need to allocate it first.

Reply:
Done.